patterns/dxp/aspect_oriented_programming.md

```diff
@@ -0,0 +1,53 @@
+# Aspect Oriented Programming guide of use
+
+> **NOTE:** This plugin was originally in metal-plugins but now lives in portal so we can provide a better developing experience.
+
+AOP provides tools to improve control on what is executed before and after other method is invoked. You can finde this component in `frontend-js-web` module in portal.
+
+See more about Aspect Oriented Programming in [https://medium.com/@kyuwoo.choi/sneak-peek-to-javascript-aop-16458f807842](this article)
+
+## Preparing the use
+
+First of all the `frontend-js-web` dependency shall be added to the `package.json` of the module in which we'll use AOP. Be sure that the version matches with the one in `frontend-js-web/bnd.bnd` file.
+
+```
+"dependencies": {
+	...
+	"frontend-js-web": "4.0.0",
+	...
+},
+```
+
+## Use case: Change the return value of an instance method invocation
+
+```
+import {AOP} from `frontend-js-web`;
+
+const foo = new Foo();
+console.log(foo.bar()); // "bar"
```

Comment:
I'd leave out the `console.log()` in the examples as they increase the clutter. (Feel free to keep them though if you think it's better.)